### ISSUE TITLE
code_coverage_analyzer.rb:160:in `aggregate_data': You have a nil object when you didn't expect it! (NoMethodError)

### ISSUE THREAD

**Issue by kkaempf** [relevance/rcov#30]
I hit $title during coverage testing. Looks like dest[i] is nil in line 160

**Comment by kkaempf** [comment]
This seems similar to issue #23

**Comment by kkaempf** [comment]
This happens despite deleting old coverage data
 (-> http://github.com/relevance/rcov/issues/closed#issue/23/comment/32439)

**Comment by ghost** [comment]
This issue is fixed in version 0.9.3.  To my knowlegedge github still isn't building gems, so you may have to install it manually to get the fix until then.  

git clone git://github.com/relevance/rcov.git
gem build rcov.gemspec
gem install rcov-0.9.3.gem

**Comment by kkaempf** [comment]
Well, thats exactly what I did before I ran into this bug.

**Comment by ghost** [comment]
Can you include a code sample that produces this error?

**Comment by kkaempf** [comment]
I now inserted
  dest[i] = 0 unless dest[i]
and rcov doesn't crash any more

**Comment by ghost** [comment]
Where?  Can you please be a little more specific?

**Comment by kkaempf** [comment]
I changed
    cov_arr.each_with_index{|x,i| dest[i] += x.to_i}
to

```
cov_arr.each_with_index do |x,i|
  dest[i] = 0 unless dest[i]
  dest[i] += x.to_i
end
```

at lib/rcov/code_coverage_analyzer.rb:160

(see commit 0d4ae40cd184ba7cc21ad2163b644945aa0682e0 which touched exactly this line)

**Comment by kkaempf** [comment]
Sample code will be hard since its a rather large project
(http://en.opensuse.org/YaST/Web)

The source is available as
    git clone git://git.opensuse.org/projects/yast/rest-service.git

**Comment by ghost** [comment]
great thanks!  I will get this fixed this week

**Comment by kkaempf** [comment]
A more Ruby'ic way would probably be:
    cov_arr.each_with_index do |x,i|
      dest[i] ||= 0
      dest[i] += x.to_i
    end

**Comment by abedra** [comment]
This issue has been resolved in 0.9.6.  Remember that RCov has moved to gemcutter (http://gemcutter.org) as the canonical rcov.

gem install rcov -s http://gemcutter.org
